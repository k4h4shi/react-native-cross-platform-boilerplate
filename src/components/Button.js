// @flow
import React, { Component } from "react";
import { StyleSheet, TouchableOpacity, Text } from "react-native";

type Props = {
  title: string,
  onPress: () => {}
};

export default ({ title, onPress }: Props) => (
  <TouchableOpacity onPress={onPress} style={styles.button}>
    <Text style={styles.title}>{title}</Text>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  button: {
    alignItems: "center",
    justifyContent: "center"
  },
  title: {
    backgroundColor: "transparent",
    borderStyle: "solid",
    borderColor: "black",
    borderRadius: 3,
    borderWidth: 1.5,
    padding: 6,
    color: "black"
  }
});
