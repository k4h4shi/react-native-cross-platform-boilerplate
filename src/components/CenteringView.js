// @flow
import React, { Component } from "react";
import { StyleSheet, View } from "react-native";

type Props = { children: {} };

export default ({ children }: Props) => (
  <View style={styles.centering}>{children}</View>
);

const styles = StyleSheet.create({
  centering: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center"
  }
});
