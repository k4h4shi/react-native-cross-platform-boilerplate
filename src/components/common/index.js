export { default as Button } from "./Button";
export { default as CenteringView } from "./CenteringView";
