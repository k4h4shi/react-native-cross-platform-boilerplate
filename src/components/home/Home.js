// @flow
import React, { Component } from "react";
import { Text } from "react-native";
import { Button, CenteringView } from "../../components/common";

type Props = {
  onPressButton: () => void
};

const Home = ({ onPressButton }: Props) => (
  <CenteringView>
    <Text>Home Screen</Text>
    <Button title="Show samples" onPress={onPressButton} />
  </CenteringView>
);

export default Home;
