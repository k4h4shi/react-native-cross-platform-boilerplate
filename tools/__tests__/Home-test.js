// __tests__/App-test.js
import React from "react";
import { Home } from "../../src/components/home";

import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
