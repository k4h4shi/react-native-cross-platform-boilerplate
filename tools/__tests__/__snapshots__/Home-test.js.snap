// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Home Screen
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "black",
          "borderRadius": 3,
          "borderStyle": "solid",
          "borderWidth": 1.5,
          "color": "black",
          "padding": 6,
        }
      }
    >
      Show samples
    </Text>
  </View>
</View>
`;
